go/vt/schema/online_ddl.go

```diff
@@ -21,18 +21,17 @@ import (
 	"encoding/json"
 	"fmt"
 	"regexp"
-	"strings"
 	"time"
 
-	"github.com/google/uuid"
-
 	"vitess.io/vitess/go/vt/topo"
 )
 
 var (
-	migrationBasePath    = "schema-migration"
-	onlineDdlUUIDRegexp  = regexp.MustCompile(`^[0-f]{8}_[0-f]{4}_[0-f]{4}_[0-f]{4}_[0-f]{12}$`)
-	strategyParserRegexp = regexp.MustCompile(`^([\S]+)\s+(.*)$`)
+	migrationBasePath                 = "schema-migration"
+	onlineDdlUUIDRegexp               = regexp.MustCompile(`^[0-f]{8}_[0-f]{4}_[0-f]{4}_[0-f]{4}_[0-f]{12}$`)
+	strategyParserRegexp              = regexp.MustCompile(`^([\S]+)\s+(.*)$`)
+	onlineDDLGeneratedTableNameRegexp = regexp.MustCompile(`^_[0-f]{8}_[0-f]{4}_[0-f]{4}_[0-f]{4}_[0-f]{12}_([0-9]{14})_(gho|ghc|del|new)$`)
+	ptOSCGeneratedTableNameRegexp     = regexp.MustCompile(`^_.*_old$`)
```

Comment:
I was a bit unsure at first about ignoring _.*_old tables since it looks like it has a higher probability of name clashes. But on the other hand it might be a cool VReplication feature: you can use this naming convention for deprecated tables which results in movetables/reshard automatically ignoring tables!

Reply:
:+1: and unfortunately pt-osc does not let you control this specific name; it always renames `your_table` to `_your_table_old` (with potential extra `_` prefixes like `____your_table_old`)

Reply:
I'm interested in `onlineDDLGeneratedTableNameRegexp` and VStream API. If we can't upgrade Vitess to this version in production yet. Can I use `onlineDDLGeneratedTableNameRegexp` to explicitly filter out "gh-ost" internal tables from the VStream API, in order to avoid this type of schema reload issues? Thanks

Reply:
Yes, that regexp will work correctly. I assume this is a temporary solution for you.

Reply:
Thanks. Yes, it'll be a temporary solution until we upgrade our vitess to `v9.0.0-rc1`, which includes the feature this PR provides.